### ISSUE TITLE
MAINTAINER TEAM definition for different applications

### ISSUE THREAD

**Issue by AlejandroCornejo** [KratosMultiphysics/Kratos#8695]
**Description**
Dear @KratosMultiphysics/all 

We discussed this with @RiccardoRossi and we think that would be important/useful/transparent to have a **team of maintainer developers** assigned or attached to a certain application.

The objective of this issue is to know what is your opinion regarding this idea and,  if you agree, defining these teams.

Personally, I can be a maintainer of the `StructuralMechanicsApp` and especially in the `ConstitutiveLawsApp`.  Also the `FemToDemApp `since I'm the only developer there.

Glad to hear you opinions,

I know that @marandra is interested in being inside the `StructuralMechanicsApp `

**Comment by dcagri** [comment]
I think this is a great idea!
I'm ready to be assigned for the PFEM2Application. I believe that @miguelmaso could be interested as well.
Regards

**Comment by sunethwarna** [comment]
What is the difference between **team of maintainer developers** and `CODEOWNERS`?

**Comment by AlejandroCornejo** [comment]
> 
> 
> What is the difference between **team of maintainer developers** and `CODEOWNERS`?

I would say (@RiccardoRossi  maybe has something to say) that a `CODEOWNER` is someone that is using the App and may be interested in the updates of it. 

A maintainer is more involved in the sense that he could address the requirements/bugs/updating of the App. It has more "responsibility" in some sense. He should be also the direct contact if some external party wants more info.

**Comment by loumalouomega** [comment]
Also structural, but I don't have that much time

**Comment by RiccardoRossi** [comment]
Dear all,
    to be honest i am a bit lost about the exact definition that should be used, i mean if CODEOWNER,MANTAINER or something else. What is really important to me is that someone steps up in taking the responsability (and the honor, of course!) of mantaining the code...

**Comment by maceligueta** [comment]
> > > What is the difference between **team of maintainer developers** and `CODEOWNERS`?
> > 
> > 
> > I would say (@RiccardoRossi maybe has something to say) that a `CODEOWNER` is someone that is using the App and may be interested in the updates of it.
> > A maintainer is more involved in the sense that he could address the requirements/bugs/updating of the App. It has more "responsibility" in some sense. He should be also the direct contact if some external party wants more info.
> 
> Indeed it is completely opposite. `CODEOWNER`(s) gives kind of complete control over the changes in one application. That is to say that no change can be merged without their explicit approval. For instance, this is what happens with the `KratosCore`, which is "owned" by the @KratosMultiphysics/technical-committee (this is why it is always required to approve the changes by one of the 5 members of the KTC).
> 
> This means that one can be maintainer of one application without explicitly being a `CODEOWNER`, even though I'd expect all the official maintainers of one application to become `CODEOWNER`(s) as soon as we start using this feature not only in the core but in the applications.
> 
> Last but not least, I stress the quotes in "owned", since Kratos is "owned" by the community.

This is the first time a hear about this definition of CODEOWNERS. I am aware that cod

**Comment by maceligueta** [comment]
I was not aware that the code owners were the only ones who could approve the PR, as @rubenzorrilla is explaining. I thought anyone could approve any PR.

**Comment by maceligueta** [comment]
I can be blamed for any problem with the DEM Application.

**Comment by rubenzorrilla** [comment]
[This](https://docs.github.com/es/github/creating-cloning-and-archiving-repositories/about-code-owners) is what I meant. So far this mechanism is only active in the KratosCore (this is why a review request is automatically assigned to the KTC).

**Comment by maceligueta** [comment]
Apparently, there’s the option to automatically just assign the owners as reviewers and the option of actually requiring a review by the owners. I always thought we were in the first case, but I actually think that the second option is better.
https://github.community/t/codeowners-assigned-for-review-versus-required-for-review/162087

**Comment by philbucher** [comment]
Regarding CODEOWNERS check this: https://github.com/KratosMultiphysics/Kratos/pull/6545#issuecomment-602638990

We decided not to activate the review by codeowner as it was considered to add even more bureaucracy to merging PRs.

**Comment by philbucher** [comment]
BTW a large number of apps (including DEM) is already in the [CODEOWNERS file](https://github.com/KratosMultiphysics/Kratos/blob/master/.github/CODEOWNERS)

**Comment by philbucher** [comment]
@AlejandroCornejo what is the difference of this discussion from #7814?

**Comment by AlejandroCornejo** [comment]
> 
> 
> @AlejandroCornejo what is the difference of this discussion from #7814?

Essencially is the same hehehe

**Comment by jginternational** [comment]
Are the `CODEOWNERS` also responsible of the GUI ? 
Having it updated and working with the kratos master branch features? Or at least documenting the work to be done as issues in the [GiDInterface](https://github.com/KratosMultiphysics/GiDInterface) repo

**Comment by rubenzorrilla** [comment]
> Are the `CODEOWNERS` also responsible of the GUI ?
> Having it updated and working with the kratos master branch features? Or at least documenting the work to be done as issues in the [GiDInterface](https://github.com/KratosMultiphysics/GiDInterface) repo

If there are official and supported GUIs available, in my opinion `CODEOWNERS` should be responsible of these too.

**Comment by jginternational** [comment]
> > Are the `CODEOWNERS` also responsible of the GUI ?
> > Having it updated and working with the kratos master branch features? Or at least documenting the work to be done as issues in the [GiDInterface](https://github.com/KratosMultiphysics/GiDInterface) repo
> 
> If there are official and supported GUIs available, in my opinion `CODEOWNERS` should be responsible of these too.

Perfect! 


Happy Star Wars day !!

https://user-images.githubusercontent.com/5918085/117018582-e93cb680-acf4-11eb-8a15-660585dfa90c.mp4

**Comment by AlejandroCornejo** [comment]
I am going to put a tentative list of Apps and its maintainers, please add yours :D

- `StructuralMechanicsApp`: @AlejandroCornejo @marandra @RiccardoRossi 
- `ConstitutiveLawsApp`: @AlejandroCornejo @marandra 
- `ContactStructuralMechanicsApp`: @loumalouomega @AlejandroCornejo 
- `FluidDynamicsApp`: @rubenzorrilla 
- `FemToDemApp`: @AlejandroCornejo 
- `DEMApplication`: @maceligueta 
- `DEMStructuresApplication`: @maceligueta 
- `PoroMechanics`: @ipouplana 
- `DamApplication`: @ipouplana 
- `SwimmingDEMApplication`: @GuillermoCasas 
- `PfemFluidDynamicsApp`: @AFranci @rlrangel

**Comment by dcagri** [comment]
@miguelmaso do you agree?

**Comment by philbucher** [comment]
I still don't fully understand why we need separate teams other than the app teams
Consider that the app teams already contain the ppl that you listed as maintainers. 
So why do we need to create (even more) bureaucracy?

**Comment by AlejandroCornejo** [comment]
> 
> 
> I still don't fully understand why we need separate teams other than the app teams
> Consider that the app teams already contain the ppl that you listed as maintainers.
> So why do we need to create (even more) bureaucracy

I mean we do not care if the name is `Maintainer `or using the old `CODEOWNER`. The important thing is to define clearly who is in charge of updating stuff, maintain the app (bugs) and be the contact person in case of doubts/collaboration, etc...

Now, the `CODEONWER `is informed of the added code, **but who is going to solve an issue if a bug arises?** this is for me the important point

**Comment by philbucher** [comment]
I am not sure if this helps. After all this is an open source project, we do not employ people to write code, we can only encourage them and make simplify contributing
And by adding more an more bureaucracy we make it more and more difficult.

I agree that it makes sense to have someone to go to when a bug is discovered, but here I think it is beneficial if there is a large team behind an app, as usually there is someone around that has the time and resources to fix it.

**Comment by AlejandroCornejo** [comment]
Yes, I agree with you, I'd rephrase as " but who is going to **manage** an issue if a bug arises?"

**Comment by philbucher** [comment]
usually the devs of the team that have time atm manage it

**Comment by AlejandroCornejo** [comment]
Honestly I created this after speaking with @RiccardoRossi, maybe he has more insight. If it works as it is, ok for me 👍

**Comment by RiccardoRossi** [comment]
as i told my main intention was to tell that we need to commit ourselves to mantaining the StructuralMechanics more actively.
I didn't intend to enter in the semantic discussion between codeowners/mantainers etc... i am completely open to your suggestions about that

**Comment by rubenzorrilla** [comment]
Why don't we add this to one of the next community meetings? I think this are the kind of things to be discussed in there.

**Comment by maceligueta** [comment]
I will post the question as soon as we anounce it.

**Comment by pooyan-dadvand** [comment]
After reading and thinking about it all these days, I think that I like more the maintainers and not code owners. Just to add that I consider myself a maintainer of the core but not a code owner.  

I think the mechanism of the code owner for having automatic notification is useful but I would not extend its literal meaning to foster territories inside the project. 

Nevertheless, I would extend the application maintainers to **application managers** because I think they should be the ones who define the pace of their application rather than only maintain the application.

**Comment by AlejandroCornejo** [comment]
What's the status of this? :)
